PWGMM/Rivet/Tools/hepmc.py

```diff
@@ -0,0 +1,2642 @@
+            return self._lineno
+
+        # ------------------------------------------------------------
+        @property
+        def version(self):
+            """Version of HepMC encoder"""
+            return self._reader.version
+
+        # ----------------------------------------------------------------
+        @property
+        def attributes(self):
+            """Get dictionary of run attributes (possibly empty)"""
+            return self._reader.attributes
+
+        # ----------------------------------------------------------------
+        @property
+        def weights(self):
+            """Get dictionary of run weight names (possibly None)"""
+            return self._reader.weights
+
+        # ----------------------------------------------------------------
+        def read(self, check=True):
+            """Read a single event from stream
+
+            Returns
+            -------
+            event : dict
+                Read event or None
+            """
+            self._lineno, ev = self._reader.read(self._stream, self._lineno, check)  # X  # X  # X  # Leave me  alone
+
+            return ev
+
+    # ----------------------------------------------------------------
+    def __init__(self, inp, check=True):
+        """Context mananger of HepMC input.
+
+        Paramters
+        ---------
+        inp : io.TextIObase or str
+            File to read from or name of file
+            If a file, use as is.  If a string, open either as text file or
+            gzipped text file
+        """
+        self._lineno = 0
+        self._stream = inp
+        self._own = False
+        self._check = check
+
+        if isinstance(inp, str):
+            with open(inp, "rb") as tmp:
+                magik = [tmp.read(1), tmp.read(1)]
+                gzipped = magik[0] == b"\x1f" and magik[1] == b"\x8b"
+
+            self._own = True
+            if gzipped:
+                from gzip import open as gzopen
+
+                self._stream = gzopen(inp, "rt", encoding="utf-8")
+            else:
+                self._stream = open(inp, "r")
+
+    # ----------------------------------------------------------------
+    def __iter__(self):
+        """Return event iterator"""
+        return HepMCInput.EventIterator(self._stream, self._check)
+
+    # ----------------------------------------------------------------
+    def __enter__(self):
+        """Enter context, returns event iterator"""
+        return iter(self)
+
+    # ----------------------------------------------------------------
+    def __exit__(self, *exc):
+        """Exit context, returns None"""
+        if self._own:
+            self._stream.close()
+
+        tpe, val, tb = exc
+        if val is not None:
+            from traceback import print_exception
+
+            print_exception(tpe, val, tb)
+
+        return None
+
+
+# ====================================================================
+# God, the linter screwed this one up good!
```

Comment:
you know, sarcasm translates very poorly to text, but I got your meaning.  I hope you also got mine.  

But, please _please_ do not make contributing to `O2Physics` etc. any more harder than it needs to be.   I will _strongly_ encourage you to relax the various linter and formatter rules so that perfectly reasonably formatted code will pass without any hick-ups.   You can still issue warnings if you like, but don't make people adhere to a set of more or less ill-informed arbitrary rules.   

Adherence to  formatting and linting rules _does not_ guarantee sane or good code - only well-thought-out code does that.  Formatting and linting rules only makes it more cumbersome to contribute with no _direct_ benefit for the code or the developers.